server/src/main/java/com/cloud/storage/StorageManagerImpl.java

```diff
@@ -1148,6 +1150,91 @@ public PrimaryDataStoreInfo updateStoragePool(UpdateStoragePoolCmd cmd) throws I
         return (PrimaryDataStoreInfo)_dataStoreMgr.getDataStore(pool.getId(), DataStoreRole.Primary);
     }
 
+    @Override
+    public boolean changeStoragePoolScope(ChangeStoragePoolScopeCmd cmd) throws IllegalArgumentException, InvalidParameterValueException, PermissionDeniedException {
+        Long id = cmd.getId();
+
+        Long accountId = cmd.getEntityOwnerId();
+        if (!_accountMgr.isRootAdmin(accountId)) {
+            throw new PermissionDeniedException("Only root admin can perform this operation");
+        }
+
+        ScopeType scopeType = ScopeType.validateAndGetScopeType(cmd.getScope());
```

Comment:
No, it can't be.
I can see how usage of validateAndGetScopeType can cause confusion about the assumption.
Have changed the code to use  EnumUtils.getEnumIgnoreCase to get the enum from string.